File: Source/FrontEnd/Portal.Web/Cdn/js/angular/editorspick/video/video.js
// Copyright (c) Clickberry, Inc. All rights reserved.
// Licensed under the GNU GENERAL PUBLIC LICENSE Version 3. See License.txt in the project root for license information.

angular.module("ngClickberry.editorspick.video", [
        'ngClickberry.video',
        'directives.popup',
        'directives.player',
        'directives.socialshare',
        'directives.loader',
        'services.watch',
        'services.video.comments',
        'filters',
        'ui.router'
    ])

// Routes
    .config([
        "$stateProvider", "userRoles", function($stateProvider, userRoles) {
            $stateProvider
                .state('portal.editorspick.video', {
                    url: '/{id}',
                    data: {
                        pageTitle: 'Video on Clickberry Online Storage',
                        authorizedRoles: [userRoles.all]
                    },
                    views: {
                        'video': { // reusing template and controller from ngClickberry.video module
                            controller: 'VideoCtrl',
                            templateUrl: 'portal.video.html'
                        }
                    }
                });
        }
    ]);

File: Source/Entities/Portal.DTO/Projects/Video.cs
// Copyright (c) Clickberry, Inc. All rights reserved.
// Licensed under the GNU GENERAL PUBLIC LICENSE Version 3. See License.txt in the project root for license information.

using System;

namespace Portal.DTO.Projects
{
    public sealed class Video
    {
        /// <summary>
        ///     Gets or sets a video file uri.
        /// </summary>
        public string Uri { get; set; }

        /// <summary>
        ///     Gets or sets a content type.
        /// </summary>
        public string ContentType { get; set; }

        /// <summary>
        ///     Gets or sets a video width.
        /// </summary>
        public int Width { get; set; }

        /// <summary>
        ///     Gets or sets a video height.
        /// </summary>
        public int Height { get; set; }

        /// <summary>
        ///     Gets or sets a creation date.
        /// </summary>
        public DateTime Created { get; set; }
    }
}